### PR TITLE
Align info.description Auth section with ICM

### LINKED ISSUE #130: Align info.description Auth section with ICM 0.2.0
**Problem description**

Text get replaced by the latest text, best the one from upcoming r0.2.0-rc.2 of ICM, which got merged within https://github.com/camaraproject/IdentityAndConsentManagement/pull/186 and is currently available in main branch of ICM.

**Expected action**

Update the info.description Auth section accordingly

### PR DESCRIPTION
**Pull request by fernandopradocabrillo**
#### What type of PR is this?

Add one of the following kinds:
* documentation

#### What this PR does / why we need it:
Align Authorization and Authentication info.description section with current ICM approved text in v0.2.0

#### Which issue(s) this PR fixes:

Fixes #130

### CODE REVIEW

File: code/API_definitions/number_verification.yaml [hunk 1]
```diff
@@ -40,11 +40,11 @@ info:
...
+    Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.
 
-    In the case of the Number Verification API scenario and according to the API definition, 3-legged access tokens must be used by API clients to invoke this API with dedicated scope. The API client must authenticate on behalf of a specific user to use this service. This must be done via mobile network authentication.
+    It is important to remark that in cases where personal user data is processed by the API, and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of 3-legged access tokens becomes mandatory. This measure ensures that the API remains in strict compliance with user privacy preferences and regulatory obligations, upholding the principles of transparency and user-centric data control.
```
Comment (hdamker) [inline code/API_definitions/number_verification.yaml:47]: The line 47 shall not be deleted, as it is specific to Number Verification (this API wouldn't work at all with 2-legged authentication, as the mobile network authentication during the authentication phase is the essential part of the API functionality).

Comment (hdamker) [reply code/API_definitions/number_verification.yaml:47]: Beyond that I would ask @AxelNennker here for the review.

Comment (fernandopradocabrillo) [reply code/API_definitions/number_verification.yaml]: True! I've reversed the change of that line. Thanks

**Review by hdamker — APPROVED** [review]
LGTM

**Review by AxelNennker — APPROVED** [review]

**Pull request merged**